### ISSUE TITLE
It's a little hard to sleep on... (tents, improvised shelters). A potential sleep expansion later down the line?

### ISSUE THREAD

**Issue by Learner10** [CleverRaven/Cataclysm-DDA#9191]
There are a couple of issues out there around sleeping.
Lets say you set up a tent (large or small) in a field, or on a road. Whenever you try to sleep inside of it, the game will think you're sleeping on whatever the tent was placed on, rather then the tent's ground-sheet. This results in you getting the message somewhere along the lines of "You find it a little difficult to sleep on the dirt/grass/pavement". Obviously, it shouldn't be comfortable, but despite that, the fact that you're at least sleeping on something softer/cleaner should make sleeping in tents... at least neutral.
To compare, sleeping on a 'hard' floor on the inside of a building is neutral territory when trying to sleep. It's not uncomfortable, but it's not going to be soft either.
This similar issue also brings us over to problem number 2. Improvised shelters. They too have a similar issue, with the exception that sleeping in (read: on), an improvised shelter, is also uncomfortable. (The exact message is "It's a little hard to get to sleep on this improvised shelter." On it says.

I propose that the easiest I.E. early solution is to improve the sleeping quality of groundsheet and improvised shelters to neutral.

The final solution for much later down the line, seems like an upgrade to the sleep system. Something along the lines of having 4-5 levels of comfort when trying to sleep (along with alternative messages/thoughts conveying the comfort of what you're sleeping on): "This is downright unpleasant" (pavement, dirt, stone/rocks; could result on having slept on something the wrong way, or morale decrease), "It's not the worst" (grass, 'maybe' roll mats, and improvised shelters), 'maybe' our current neutral rank (I suppose groundsheets and carpeted flooring), "Certainly beats the floor" (couches, car seats, armchairs, essentially soft furniture), and "You could sleep here for days/hours" (where beds would obviously go.)

Something to consider if we do expand on the sleep system. We'd have to consider how pillows, sleeping bags, and sheets would modify the comfort level of things we sleep on. (You'd think that I'd be able to get at least a neutral rank of sleeping comfort if I were sleeping inside a tent wearing a sleeping bag...)

To summarize: Groundsheet & improvised shelters don't do anything to how easily you can sleep. It seems just as hard to sleep inside a tent as it is to sleep on the grass outside.
##
